### ISSUE TITLE
Autocomplete support for keyboard navigation

### ISSUE THREAD

**Issue by jffaust** [stsrki/Blazorise#1320]
Right now, the only way to use the autocomplete component without the mouse is to type the text until there's only one option suggested and then press Tab or Enter.
It would be very useful to be able to use the up and down arrow keys to change the selection of the autocomplete component once a couple options are shown. 

Here's a jsfiddle that I found that contains the behavior I would like implemented: http://jsfiddle.net/rniemeyer/kEdT5/

If you type the letter "f", two options show up and then you can use the up and down arrow keys to change the selection. Although this example changes the content in the input box when we change the selection in the dropdown with the arrow keys, I would personally prefer that the input box keeps the "f" content until Tab or Enter is pressed when the option that the user wants is selected.

Maybe this is possible right now but looking at the docs and trying a couple things, I haven't been able to make it work.
Thanks for looking into this :)

**Comment by stsrki** [comment]
It's already possible. I just tried it to confirm and it's working just fine. You need to write at least one letter to show a couple of options and then select it with up and down keys.

**Comment by jffaust** [comment]
@stsrki Any ideas why I'm not able to make it work?
Here's a gif I made trying to use the up and down arrow keys (along with other keys). I end up using Tab at the end and it selects the first option in the list.

![autocomplete](https://user-images.githubusercontent.com/4367222/93724168-ad444f00-fb72-11ea-8405-490497c87a2d.gif)

Could it be because I'm using the autocomplete component in a modal?

Thanks for all the help, it is much appreciated 👍

**Comment by jffaust** [comment]
I tried with Edge, Firefox and Opera. Doesn't seem to be a browser issue

**Comment by stsrki** [comment]
That's strange. For me, it works. I just tried to place it on modal and with different browsers and no issues what so ever. Maybe you're missing some of `blazorise.js` files?

**Comment by jffaust** [comment]
Here's my `index.html`: 
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"
    />
    <title>Flow</title>
    <base href="/" />

    <link href="css/app.css" rel="stylesheet" />
    <link href="manifest.json" rel="manifest" />
    <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png" />
    <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png" />
    <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png" />
    <link rel="manifest" href="/site.webmanifest" />
    <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5" />
    <meta name="msapplication-TileColor" content="#da532c" />
    <meta name="theme-color" content="#ffffff" />

    <link href="css/material.min.css" rel="stylesheet" />

    <link
      href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,500,500i,700,700i|Roboto+Mono:300,400,700|Roboto+Slab:300,400,700"
      rel="stylesheet"
    />
    <link
      href="https://fonts.googleapis.com/icon?family=Material+Icons"
      rel="stylesheet"
    />

    <link href="_content/Blazorise/blazorise.css" rel="stylesheet" />
    <link
      href="_content/Blazorise.Snackbar/blazorise.snackbar.css"
      rel="stylesheet"
    />
    <link
      href="_content/Blazorise.Material/blazorise.material.css"
      rel="stylesheet"
    />

    <!-- Optional JavaScript -->
    <!-- jQuery first, then Popper.js, then Material JS -->
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js"></script>
    <script src="js/material.min.js"></script>

    <script src="_content/Blazorise/blazorise.js"></script>
    <script src="_content/Blazorise.Material/blazorise.material.js"></script>
  </head>

  <body>
    <app>Loading...</app>

    <!-- <div id="blazor-error-ui">
      An unhandled error has occurred.
      <a href="" class="reload">Reload</a>
      <a class="dismiss">🗙</a>
    </div> -->
    <script src="_framework/blazor.webassembly.js"></script>
    <script>
      navigator.serviceWorker.register("service-worker.js");
    </script>
  </body>
</html>
```

Don't think I'm missing any js files, according to the documentation.

**Comment by jffaust** [comment]
I thought it might be because I'm using a keyboard that doesn't have dedicated arrow keys and I'm using the arrow keys from a numpad only but I also tried with an on-screen keyboard and it's still not working :(

@stsrki What would be the easiest way to debug this? Do you think I could somehow remove the Nuget package reference and instead directly reference a checked out version of the Blazorise.Components and then try to debug the component to see what's happening in the `HandleTextKeyDown` method of the component?

**Comment by stsrki** [comment]
The order and placement of JS files is sometimes critical, so take a look at [demo index.html](https://github.com/stsrki/Blazorise/blob/master/Demos/Blazorise.Demo.Material/wwwroot/index.html) and update your code accordingly. JS files must be place inside of body, after the app tag and before `_framework/blazor.webassembly.js`

About the debugging. The best option is to probably clone the repo and add a reference to your project.

**Comment by jffaust** [comment]
I used a project reference and did some debugging and the code was executing as normal. I then did some test and using the arrows then tabbing and I was able to select the option I wanted, I just had no visual clue of which option was selected. Looking at the html, I did see the `active` class moving from one option to another, I just didn't have the same css applied as when hovering on an option (which is what really threw me off initially).

That's what I get for having too many custom css rules, but I can't move away from my own dark theme :P
